Libraries/LibMarkdown/Text.h

```diff
@@ -51,7 +52,7 @@ class Text final {
     String render_to_html() const;
     String render_for_terminal() const;
 
-    bool parse(const StringView&);
+    static OwnPtr<Text> parse(const StringView&);
```

Comment:
Can't `Text` be simply returned by value? (as an `Optiomal<Text>` — and make sure to `move()` it where appropriate) Unlike, the other types, it's not a `Block` so we don't need it to be behind a pointer for that reason.

Reply:
Sure, I can return it as an `Optional<Text>`, I just am not sure at all when to use smart pointers or not. My other question is about `move()`, when would it be appropriate to move? Apologies for the questions, but I've never worked with C++ before.

Reply:
In case of LibMarkdown specifically, it makes sense to return subclasses of `Block` by pointer (so, by some smart pointer), because they have virtual methods, and you want to store them (multiple blocks of different concrete types) in a `Vector<SomePtr<Block>>`.

But `Text` is not a block, although it has a similar API to one. So no need for pointers. Though there's nothing preventing you from using a pointer with it, it's just unnecessary.

If you start passing the text by value, use `move()` basically anywhere you're moving it 🙂 to prevent implicit copying (yay C++). So, when you're constructing an `Optional` out of it, and when you're extracting it from the optional, and so on.

Or maybe scratch the optional, even — `Text` is designed in a way that parsing to it can never fail.

Reply:
Oh, it will help if you add `AK_MAKE_NONCOPYABLE` to `Text`.

Reply:
Ah thank you very much, that’s super helpful. Ill make these changes the next chance I get!

Reply:
I believe that my latest changes are what you intended, though I'm still not super sure, so please let me know :)